main/src/ca/uwaterloo/flix/runtime/shell/ShellParser.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 Paul Butcher
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package ca.uwaterloo.flix.runtime.shell
+
+import org.jline.reader.{EOFError, ParsedLine, Parser}
+import org.jline.reader.Parser.ParseContext
+
+import scala.jdk.CollectionConverters._
+
+/**
+  * Minimal implementation of `ParsedLine`, necessary to keep jline happy
+  */
+class Parsed(s: String) extends ParsedLine {
+
+  def wordIndex(): Int = -1
+  def word() = ""
+  def wordCursor(): Int = -1
+  def words() = Nil.asJava
+  def cursor() = -1
+  def line() = s
+}
```

Comment:
All of the above look a bit scary. I hope they cannot lead to IndexOutOfBounds?

Reply:
Does JLIne have some documentation that we could borrow here?

Reply:
The honest answer is I don't know. The jline documentation is frustratingly sparse so I've worked most of this out by reading the jline code and the code of other projects that make use of jline.

However, I *think* that none of the above is ever actually used. It's used by the highlighting code within jline, and the code which performs line continuation if there are unclosed quotes, brackets, etc. Because we have all the above switched off, we just need to return something (anything) that conforms to the `ParsedLine` interface.

At some point in the future, if we could populate the above with something derived from the parboiled parser, then we could switch on syntax highlighting etc. within jline, which might be nice.

I'll add a comment to the above effect, and a link to the relevant portion of the jline source.

Reply:
OK. Thanks for the information 👍